src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -582,9 +597,9 @@ static inline struct mlx5_grh_av *uct_dc_mlx5_ep_get_grh(uct_dc_mlx5_ep_t *ep)
  */
 #define UCT_DC_MLX5_CHECK_RES(_iface, _ep) \
     { \
+        UCT_DC_MLX5_CHECK_DCI_RES(_iface, _ep) \
```

Comment:
regarding 2 - as we discussed offline - what happens if ep without dci is added to tx arbiter? it may not be able to allocate a dci for long time because the dci-wait arbiter dispatch will take first any available dci.

Reply:
right, in this case we could try to move the element to waitq arbiter if pending callback returns `NO_RESOURCES` and all dcis are taken.

Reply:
maybe this could cause a situation the pending request moves back and forth between arbiters? IMO better have some well defined order of scheduling to prevent this